src/data-workspace/validation/use-validation-result.js

```diff
@@ -0,0 +1,118 @@
+import { useCallback } from 'react'
+import { useIsMutating, useQueries } from 'react-query'
+import {
+    useDataSetId,
+    useOrgUnitId,
+    usePeriodId,
+} from '../../context-selection/index.js'
+import { useApiAttributeParams } from '../../shared/index.js'
+
+const getValidationMetaDataQueryKey = (datasetId) => {
+    const queryKey = [
+        `validationRules`,
+        {
+            params: {
+                dataSet: datasetId,
+                fields: [
+                    'id',
+                    'importance',
+                    'operator',
+                    'leftSide',
+                    'rightSide',
+                    'displayInstruction',
+                    'displayDescription',
+                    'displayName',
+                ],
+            },
+        },
+    ]
+
+    return queryKey
+}
+
+const getValidationQueryKey = (
+    datasetId,
+    { periodId, orgUnitId, categoryComboId, categoryOptionIds }
+) => {
+    const params = {
+        pe: periodId,
+        ou: orgUnitId,
+        cc: categoryComboId,
+        cp: categoryOptionIds?.join(';'),
+    }
+
+    const validationQueryKey = [
+        `validation/dataSet/${datasetId}`,
+        {
+            params,
+        },
+    ]
+    return validationQueryKey
+}
+export const useValidationResult = () => {
+    const [datasetId] = useDataSetId()
+
+    const [periodId] = usePeriodId()
+    const [orgUnitId] = useOrgUnitId()
+    const {
+        attributeCombo: categoryComboId,
+        attributeOptions: categoryOptionIds,
+    } = useApiAttributeParams()
+
+    const validationQueryKey = getValidationQueryKey(datasetId, {
+        periodId,
+        orgUnitId,
+        categoryComboId,
+        categoryOptionIds,
+    })
+    const validationMetaDataQuery = getValidationMetaDataQueryKey(datasetId)
```

Comment:
Some naming suggestions in this file I think would make it easier to read:

```suggestion
    const validationMetaDataQueryKey = getValidationMetaDataQueryKey(datasetId)
```